pkg/provider/provider.go

```diff
@@ -462,14 +462,14 @@ func GetPodIPsPerNet(annotation string) (ips map[string][]string, err error) {
 	// Otherwise add all non default interfaces
 	for _, cniInterface := range cniInfo {
 		if !cniInterface.Default {
-			ips[cniInterface.Name] = cniInterface.IPs
+			ips[cniInterface.Name] = append(ips[cniInterface.Name], cniInterface)
```

Comment:
Why have a list of CniNetworkInterface here, It shoudl be a CniNetworkInterface. The List of IPs is inside the CniNetworkInterface.

Reply:
Good call I'll switch it.

Reply:
Done.